dlib/CMakeLists.txt

```diff
@@ -438,7 +441,7 @@ if (NOT TARGET dlib)
             set(CMAKE_REQUIRED_LIBRARIES ${JPEG_LIBRARY})
             CHECK_FUNCTION_EXISTS(jpeg_read_header LIBJPEG_IS_GOOD)
          endif()
-         if (JPEG_FOUND AND LIBJPEG_IS_GOOD AND NOT APPLE)
+         if (JPEG_FOUND AND LIBJPEG_IS_GOOD AND (HUNTER_ENABLE OR (NOT APPLE)))
```

Comment:
`find_package(JPEG CONFIG REQUIRED)` should be used here and imported Jpeg targets instead of `JPEG_INCLUDE_DIR` and `JPEG_LIBRARY`:
* https://docs.hunter.sh/en/latest/packages/pkg/Jpeg.html

Reply:
I guess the hunter fork has always used this.  Actually, In these cases I am inclined to replace all of the non package code with

```
hunter_add_package(Jpeg)
find_package(JPEG CONFIG REQUIRED)
target_link_libraries(... JPEG::jpeg)
```

I think trying to to support `CONFIG` and non-`CONFIG` use simultaneously makes it rather messy.  The same would be try for PNG in this code.  Thoughts?

Reply:
I'll send a separate PR with minimal `hunter_add_package() + find_package()` dependency management for comparison (+ gif).

Reply:
> I guess the hunter fork has always used this

May be this code wasn't used? Don't know.

> Thoughts?

Using Hunter API is the best option. But if tests are working I'm okay with anything. We may have issues in future with relocation and possible conflicts with system libraries as usual.

Reply:
I think this one is easier to maintain: https://github.com/hunter-packages/dlib/pull/13

Reply:
If that looks ok and there are no objections, I'll close this one in favor of #13